pages/fundamentals/data-types.mdx

```diff
@@ -772,6 +773,180 @@ and display it nicely in the client-side results.
+| :---------------:  | :----------: | :-----------: |
+|     Cartesian 2D   | 7203         |  cartesian    |
+|     Cartesian 3D   | 9157         |  cartesian-3d |
+
+
+### Point functions
+
+#### point(parameters: map) -> [null, point]
+
+Point function is used to create the point data type. It takes in a map of parameters to create the type. The point type can be specified using its `crs` or `srid`.
+If any of the parameters evaluates to null the point function returns null.
+
+- **WGS-84 2D**
+  A 2D geographic point in the **WGS-84 CRS** can be created in one of two ways:
+
+  - **longitude** and **latitude** (if these are specified, and the `crs` is not, then the `crs` is assumed to be `WGS-84`).
+  ```cypher
+  RETURN point({longitude:-73.93, latitude:40.73}) AS new_york;
+  ```
+  - **x** and **y** (in this case, the `crs` or `srid` must be specified or will be assumed to be Cartesian).
+  ```cypher
+  RETURN point({x:-73.93, y:40.73, crs:'wgs-84'}) AS new_york;
+  ```
+
+- **WGS 84 3D**
+  A 3D geographic point in the **WGS-84 CRS** can be created in one of two ways:
+
+  - **longitude**, **latitude**, and either **height** or **z** (if these are specified, and the `crs` is not, then the `crs` is assumed to be `WGS-84-3D`).
+  ```cypher
+  RETURN point({longitude:-73.93, latitude:40.73, height:10}) AS new_york;
+  RETURN point({longitude:-73.93, latitude:40.73, z:10}) AS new_york;
+  ```
+  - **x**, **y**, and **z** (in this case, the `crs` or `srid` must be specified, or will be assumed to be Cartesian-3D).
+  ```cypher
+  RETURN point({x:-73.93, y:40.73, z:10, srid:4979}) AS new_york;
+  ```
+
+- **Cartesian 2D**
+  A 2D Cartesian point can be created by providing:
+
+  - **x** and **y** (when provided, the point is assumed to reside in a two-dimensional Cartesian system).
+  ```cypher
+  RETURN point({x:0, y:1}) AS cartesian_point;
+  ```
+
+- **Cartesian 3D**
+  A 3D Cartesian point can be created by providing:
+
+  - **x**, **y**, and **z** (when provided, the point is assumed to reside in a three-dimensional Cartesian system).
+  ```cypher
+  RETURN point({x:0, y:1}) AS cartesian_point;
+  ```
+
+#### point.distance(x: [null, point], y: [null, point]) -> [null, float]
+
+The point.distance function returns the distance between two points. Distance can only be calculated between points of the same type.
+For WGS-84 points the distance is calculated in meters and is calculated using the [Haversine](https://en.wikipedia.org/wiki/Haversine_formula) formula.
+For WGS-84 3D points the Haversine formula is applied to longitude and latitude and then the Pythagorean theorem is applied between the calculated distance and the height.
+For Cartesian points, the Pythagorean theorem is applied.
+If any of the arguments of the point.distance function evaluates to null or the point types don't match then the function returns null.
+
+{<h5> Usage: </h5>}
+
+Example of the distance between London and Zagreb:
```

Comment:
Two points in WGS-84 2D?

Reply:
Yes? What do you mean?

Reply:
```suggestion
Example of the distance between two points in WGS-84 2D:
```

Reply:
It is supposed to be real life example (yes they are WGS-84 2D) . If you think this is more clear commit.